From 8f767113d5b8a2ee89cd069d54243f8b0ac9a218 Mon Sep 17 00:00:00 2001
From: Alistair Francis <alistair@alistair23.me>
Date: Wed, 2 Sep 2020 15:28:50 -0500
Subject: [PATCH] meson.build: make TLS ELF optional

USE_ELF_TLS has replaced GLX_USE_TLS so this patch is the original "make
TLS GLX optional again" patch updated to the latest mesa.

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Alistair Francis <alistair@alistair23.me>
---
 meson.build       | 2 +-
 meson_options.txt | 6 ++++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 9b4e5e28864..15eacbe6918 100644
--- a/meson.build
+++ b/meson.build
@@ -509,7 +509,7 @@ foreach platform : _platforms
 endforeach
 
 use_elf_tls = false
-if not with_platform_windows or not with_shared_glapi
+if (not with_platform_android or get_option('platform-sdk-version') >= 29) and get_option('elf-tls')
   pre_args += '-DUSE_ELF_TLS'
   use_elf_tls = true
 
diff --git a/meson_options.txt b/meson_options.txt
index fd0d99f2f9a..f994c22ac61 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -455,6 +455,12 @@ option(
   value : true,
   description : 'Enable direct rendering in GLX and EGL for DRI',
 )
+option(
+  'elf-tls',
+  type : 'boolean',
+  value : true,
+  description : 'Enable TLS support in ELF',
+)
 option(
   'prefer-iris',
   type : 'boolean',
-- 
2.34.0

